src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,59 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final byte MAX_ITEMS_NUMBER = 10;
+    private final K[] arrayKeys;
+    private final V[] arrayValues;
+    private byte index = 0;
+    private boolean nullMarker = false;
+
+    public StorageImpl() {
+        arrayValues = (V[]) new Object[MAX_ITEMS_NUMBER];
+        arrayKeys = (K[]) new Object[MAX_ITEMS_NUMBER];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (index == MAX_ITEMS_NUMBER) {
+            index = 0;
+        }
+        for (int i = 0; i < arrayKeys.length; i++) {
+            if (arrayKeys[i] != null && arrayKeys[i].equals(key)) {
+                arrayValues[i] = value;
+                index--;
```

Comment:
```suggestion
            if (arrayKeys[i] == key || arrayKeys[i] != null && arrayKeys[i].equals(key)) {
                arrayValues[i] = value;
                return;
```
If written this way, the code below can be greatly simplified to just 3 lines without any additional checks